atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/StreamStoreRenderer.java

```diff
@@ -409,26 +410,32 @@ private void loadSingleBlockToOutputStream() {
                     line(StreamValueRow, " row = ", StreamValueRow, ".of(streamId, blockId);");
                     line("try {");
                     {
-                        line("os.write(getBlock(t, row));");
+                        line("byte[] block = getBlock(t, row);");
+                        line("if (block == null) {");
+                        line("throw new SafeRuntimeException(");
+                        line("        \"Block for stream not found. This is likely due to returning a stream "
+                                + " from a transaction and attempting to use it after it has been marked as"
+                                + " unused.\",");
+                        line("        SafeArg.of(\"blockId\", row.getBlockId()),");
+                        line("        SafeArg.of(\"id\", row.getId()));");
+                        line("}");
+                        line("os.write(block);");
                     }
                     line("} catch (RuntimeException e) {");
```

Comment:
`SafeRuntimeException` is a `RuntimeException`, so if the block doesn't exist we will catch the `Block for stream not found` exception in this catch block, and generate two sets of args. It's probably rare enough that doesn't really matter, and it is plausible though does seem a bit strange / was this intended?

Reply:
This was not intended, thanks for pointing this out.

Thinking about this some more, I'm thinking it might be better/safer to leave this as is was. Changing the exception type here can have meaningful consequences for consumers. Error that would have otherwise been ignored for alerting are now considered internal errors.

Put another way, the correct behavior is probably to not wrap arbitrary `RuntimeException`, but keeping the logging here is valuable.